Echo/PolySyncGetOpt.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file PolySyncGetOpt.hpp
+ * @brief PolySync GetOpt Command Line Utility.
+ *
+ * PUBLIC_HEADER
+ */
+
+#ifndef POLYSYNCGETOPT_HPP
+#define POLYSYNCGETOPT_HPP
+
+#include <PolySyncDataModel.hpp>
+#include <getopt.h>
+
+/**
+ * @namespace polysync
+ */
+namespace polysync
+{
+
+/**
+ * @brief PolySyncGetOpt class
+ *
+ * The PolySyncGetOpt C++ class exists to build upon the C-based getopt class.
+ * Getopt parses command line input, and validates that input with options,
+ * arguments, and option-arguments.
+ */
+class PolySyncGetOpt
```

Comment:
Remove PolySync from object name.

Also, `GetOpt` isn't a very intuitive name. In a perfect world (such as ours) the name of this object would clearly reflect what it's responsible for. 

Possibilities:

1.) `ApplicationInputHandler`
2.) Ok.....#1 is all I could come up with off the bat.